File: python/surface_slice.R
library(ncdf4)

# Open the netCDF file
nc <- nc_open("wrfout.nc")

# Get the variable you need
pm25 <- ncvar_get(nc, "tr17_1", start = c(1, 1, 1, nctime), count = c(-1, -1, -1, 1))
lon <- ncvar_get(nc, "XLONG")
lat <- ncvar_get(nc, "XLAT")

# Close the netCDF file
nc_close(nc)

# Create a new netCDF file with the variable you extracted
nc_new <- nc_create("smoke.nc")
ncvar_def(nc_new, "pm25", "double", dim = list("south_north", "west_east", "bottom_top", "time"))
ncvar_put(nc_new, "pm25", pm25)
ncvar_def(nc_new, "lon", "double", dim = list("south_north", "west_east"))
ncvar_put(nc_new, "lon", lon)
ncvar_def(nc_new, "lat", "double", dim = list("south_north", "west_east"))
ncvar_put(nc_new, "lat", lat)
nc_close(nc_new)

File: python/surface_slice.py
import xarray as xr

file='wrfout.nc'
new ='smoke.nc'

print('Open the original NetCDF file',file)
ds = xr.open_dataset(file)

print('Read 2d slices from the last fram')
pm25 = ds['tr17_1'][-1,0,:,:]
lon  = ds['XLONG'][-1,:,:]
lat  = ds['XLAT'][-1,:,:]

print('Create a new dataset with 2d slices')
new_ds = xr.Dataset({'pm25':pm25, 'lat':lon, 'lat':lat})

# Remove the time dimension from the new dataset
new_ds = new_ds.squeeze(drop=True)
if 'XTIME' in new_ds:
    new_ds = new_ds.drop_vars('XTIME')

print('Create new NetCDF file',new,'and write the dataseet to it')
new_ds.to_netcdf(new)



